src/server/game/Entities/Player/Player.cpp

```diff
@@ -8577,22 +8581,25 @@ void Player::CastItemCombatSpell(Unit* target, WeaponAttackType attType, uint32
 
             if (roll_chance_f(chance))
             {
-                // Xinef: implement enchant charges
```

Comment:
Why remove enchant charges?
Leave this conditional up and declare ``Unit* target = spellInfo->IsPositive() ? this : damageInfo.GetVictim();`` up before conditional, if not this will break enchantments with charges, also you will need to declare DamageInfo on ``CastItemCombatSpell`` up, also on Player.h and on Unit.cpp declare it again on:

``     if (GetTypeId() == TYPEID_PLAYER)
           ToPlayer()->CastItemCombatSpell(victim, damageInfo->attackType, damageInfo->procVictim, damageInfo->procEx);``

Reply:
omg this is WIP and if you so smart why you don't do it? Only leech for you privat server.
and btw its another way to run enchant charges...

Reply:
Just wanted to help you out, don't get mad

Reply:
:)